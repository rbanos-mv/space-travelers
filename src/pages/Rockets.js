import { shallowEqual, useSelector } from 'react-redux';
import RocketList from '../components/RocketList';

const Rockets = () => {
  const rocketList = useSelector((state) => state.rockets, shallowEqual);
  return (
    <div className="rockets-page">
      <RocketList rockets={rocketList} />
    </div>
  );
};

export default Rockets;
