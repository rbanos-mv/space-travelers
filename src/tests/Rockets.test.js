import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../pages/Rockets';
import setupStore from './setupStore';

describe('Test the Rockets.test components', () => {
  it('renders the Rockets page correctly', () => {
    const store = setupStore(false);
    const page = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(page).toMatchSnapshot();
  });

  it('shows the button changes from "Reserve Rocket" to "Cancel Reservation" on click event', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(screen.queryByText('Cancel Reservation')).toBeFalsy();
    fireEvent.click(screen.queryByText('Reserve Rocket'));
    expect(screen.queryByText('Cancel Reservation')).toBeTruthy();
    expect(screen.queryByText('Reserve Rocket')).toBeFalsy();
  });

  it('shows the button changes from "Cancel Reservation" to "Reserve Rocket" on click event', () => {
    const store = setupStore(true);
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(screen.queryByText('Reserve Rocket')).toBeFalsy();
    fireEvent.click(screen.queryByText('Cancel Reservation'));
    expect(screen.queryByText('Reserve Rocket')).toBeTruthy();
    expect(screen.queryByText('Cancel Reservation')).toBeFalsy();
  });

  it('shows the reserved badge on click event', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(screen.queryByText('Reserved')).toBeFalsy();
    fireEvent.click(screen.queryByText('Reserve Rocket'));
    expect(screen.queryByText('Reserved')).toBeTruthy();
  });

  it('hides the reserved badge on click event', () => {
    const store = setupStore(true);
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(screen.queryByText('Reserved')).toBeTruthy();
    fireEvent.click(screen.queryByText('Cancel Reservation'));
    expect(screen.queryByText('Reserved')).toBeFalsy();
  });
});
