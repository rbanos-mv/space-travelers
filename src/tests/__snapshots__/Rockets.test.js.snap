// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Rockets.test components renders the Rockets page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rockets-page"
      >
        <ul>
          <li
            class="row rocket-card"
          >
            <img
              alt="{rocketName}"
              height="auto"
              src="https://imgur.com/DaCfMsj.jpg"
              width="250"
            />
            <div
              class="column rocket-info"
            >
              <h3
                class="rocket-name"
              >
                Falcon 1
              </h3>
              <p
                class="rocket-description"
              >
                The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
              </p>
              <button
                class="reserve-btn"
                id="1"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rockets-page"
    >
      <ul>
        <li
          class="row rocket-card"
        >
          <img
            alt="{rocketName}"
            height="auto"
            src="https://imgur.com/DaCfMsj.jpg"
            width="250"
          />
          <div
            class="column rocket-info"
          >
            <h3
              class="rocket-name"
            >
              Falcon 1
            </h3>
            <p
              class="rocket-description"
            >
              The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
            </p>
            <button
              class="reserve-btn"
              id="1"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
